FirebaseDatabase/Tests/Integration/FIRDatabaseQueryTests.m

```diff
@@ -4157,6 +4157,310 @@ - (void)testGetForNodeReturnsCorrectValue {
   WAIT_FOR(done);
 }
 
+- (void)testGetResolvesToPersistentCacheWhenOfflineAndNoListeners {
+  FIRDatabaseReference* ref = [FTestHelpers getRandomNode];
+  [self waitForCompletionOf:ref setValue:@1];
+  @try {
+    [[ref database] goOffline];
+    __block BOOL done = NO;
+    [ref getDataWithCompletionBlock:^(NSError* _Nullable error,
+                                      FIRDataSnapshot* _Nullable snapshot) {
+      XCTAssertNil(error);
+      XCTAssertEqualObjects(@1, snapshot.value);
+      done = YES;
+    }];
+    WAIT_FOR(done);
+  } @finally {
+    [[ref database] goOnline];
+  }
+}
+
+- (void)testGetResolvesToCacheWhenOnlineAndParentListener {
+  FTupleFirebase* pair = [FTestHelpers getRandomNodePair];
+  FIRDatabaseReference* writer = pair.one;
+  FIRDatabaseReference* reader = pair.two;
+  [self waitForCompletionOf:writer setValue:@{@"a" : @1}];
+  [self snapWaiter:reader
+         withBlock:^(FIRDataSnapshot* snapshot) {
+           XCTAssertEqualObjects(
+               @{@"a" : @1}, snapshot.value);
+         }];
+  __block BOOL done = NO;
+  [[reader child:@"a"]
+      getDataWithCompletionBlock:^(NSError* _Nullable error, FIRDataSnapshot* _Nullable snapshot) {
+        XCTAssertNil(error);
+        XCTAssertEqualObjects(@1, snapshot.value);
+        done = YES;
+      }];
```

Comment:
I don't think this actually tests cache though, since the application is still online.
What if:
1. We turn off persistence
2. `writer` writes to the top-level ref
3. `reader` gets the value
4. `reader` goes offline
5. Get the data at the child location ?